src/map/script.cpp

```diff
@@ -11103,10 +11103,10 @@ BUILDIN_FUNC(announce)
 		}else{
 			struct map_session_data* sd;
 
-			if( script_rid2sd(sd) )
-				bl = &sd->bl;
-			else
+			if (!script_charid2sd(9, sd))
```

Comment:
Can you restore the original order here? Inverting the "if" adds a useless "not"-operation and actually all you do is exchange the called function from script_rid2sd to script_charid2sd.